include/fmt/std.h

```diff
@@ -389,6 +391,52 @@ struct formatter<
 #endif
   }
 };
+
+namespace detail {
+
+template <typename T, typename Enable = void>
+struct has_flip : std::false_type {};
+
+template <typename T>
+struct has_flip<T, void_t<decltype(std::declval<T>().flip())>>
+    : std::true_type {};
+
+template <typename T> struct is_bit_reference_like {
+  static constexpr const bool value =
+      std::is_convertible<T, bool>::value &&
+      std::is_nothrow_assignable<T, bool>::value && has_flip<T>::value;
+};
+
+}  // namespace detail
+
+FMT_EXPORT
+template <typename BitRef, typename Char>
+struct formatter<BitRef, Char,
+                 enable_if_t<detail::is_bit_reference_like<BitRef>::value>>
+    : formatter<bool, Char> {
+  template <typename FormatContext>
+  FMT_CONSTEXPR auto format(const BitRef& v, FormatContext& ctx) const
+      -> decltype(ctx.out()) {
+    return formatter<bool, Char>::format(v, ctx);
+  }
+};
+
+#ifdef _LIBCPP_VERSION
+
+// Workaround for libc++ incompatibility with C++ standard.
+// According to the Standard, `bitset::operator[] const` returns bool.
+FMT_EXPORT
+template <typename C, typename Char>
+struct formatter<std::__bit_const_reference<C>, Char> : formatter<bool, Char> {
+  template <typename FormatContext>
+  FMT_CONSTEXPR20 auto format(const std::__bit_const_reference<C>& v,
+                              FormatContext& ctx) const -> decltype(ctx.out()) {
+    return formatter<bool, Char>::format(v, ctx);
+  }
+};
+
+#endif
```

Comment:
I think this can be replaced by a specialization of `is_bit_reference_like` for `__bit_const_reference` which seems a bit cleaner (no duplicate formatter).

Reply:
Done.